sql: TestExperimentalRelocateNonVoters failed

sql.TestExperimentalRelocateNonVoters [failed](https://tanzanite.cluster.engflow.com/invocations/default/ac2e8b9e-e9c1-4af2-b614-fad5203df362?testReportRun=2&testReportShard=15&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [c7f37e813dbd1e6d2afdb88c5e9cd986dc061299](https://github.com/cockroachdb/cockroach/commits/c7f37e813dbd1e6d2afdb88c5e9cd986dc061299):


```
=== RUN   TestExperimentalRelocateNonVoters
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestExperimentalRelocateNonVoters2217270697
    test_log_scope.go:81: use -show-logs to present logs inline
    multitenant_admin_function_test.go:1011: -- test log scope end --
test logs left over in: outputs.zip/logTestExperimentalRelocateNonVoters2217270697
--- FAIL: TestExperimentalRelocateNonVoters (9.86s)
=== RUN   TestExperimentalRelocateNonVoters/ALTER_TABLE_x_EXPERIMENTAL_RELOCATE_NONVOTERS
    multitenant_admin_function_test.go:380: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/multitenant_admin_function_test.go:380
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            				github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:81
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:60
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:42
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/multitenant_admin_function_test.go:377
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/multitenant_admin_function_test.go:1000
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/multitenant_admin_function_test.go:346
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/multitenant_admin_function_test.go:974
        	Error:      	Received unexpected error:
        	            	pq: while carrying out changes [{ADD_NON_VOTER n1,s1} {REMOVE_NON_VOTER n2,s2}]: change replicas of r83 failed: descriptor changed: [expected] r83:/Tenant/1{0/Table/48-1} [(n4,s4):1, (n5,s5):5, (n2,s2):3NON_VOTER, (n3,s3):4, next=6, gen=31] != [actual] r83:/Tenant/1{0/Table/48-1} [(n4,s4):1, (n5,s5):5, (n2,s2):3NON_VOTER, (n3,s3):4, next=7, gen=33]
        	Test:       	TestExperimentalRelocateNonVoters/ALTER_TABLE_x_EXPERIMENTAL_RELOCATE_NONVOTERS
        	Messages:   	tenant=secondary query=``ALTER TABLE t EXPERIMENTAL_RELOCATE NONVOTERS VALUES (ARRAY[1], 1);`` leaseholder=4 replicas=[2 3 4 5] voting_replicas=[3 4 5] non_voting_replicas=[2] fromReplica=2 toReplica=1
    --- FAIL: TestExperimentalRelocateNonVoters/ALTER_TABLE_x_EXPERIMENTAL_RELOCATE_NONVOTERS (9.85s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=2</code>
 - <code>shard=15</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExperimentalRelocateNonVoters.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-39416

--- a/pkg/sql/multitenant_admin_function_test.go
+++ b/pkg/sql/multitenant_admin_function_test.go
@@ -377,13 +377,27 @@ func (te tenantExpected) validate(
 	testutils.SucceedsSoon(t, func() error {
 		rows, err := runQuery()
 		if expectedErrorMessage == "" {
-			require.NoErrorf(t, err, message)
+			if err != nil {
+				return errors.WithMessagef(err, "msg=%s", message)
+			}
 			actualResults, err := sqlutils.RowsToStrMatrix(rows)
-			require.NoErrorf(t, err, message)
-			require.Equalf(t, len(expectedResults), len(actualResults), message)
+			if err != nil {
+				return errors.WithMessagef(err, "msg=%s", message)
+			}
+			if len(expectedResults) != len(actualResults) {
+				return errors.Newf(
+					"wrong number of results; %s expected=%d got=%d",
+					message, len(expectedResults), len(actualResults),
+				)
+			}
 			for i, actualRowResult := range actualResults {
 				expectedRowResult := expectedResults[i]
-				require.Equalf(t, len(expectedRowResult), len(actualRowResult), "%s row=%d\nexpected=%v\n  actual=%v", message, expectedRowResult, actualRowResult, i)
+				if len(expectedRowResult) != len(actualRowResult) {
+					return errors.Newf(
+						"wrong number of columns; %s row=%d expected=%v actual=%v",
+						message, i, len(expectedRowResult), len(actualRowResult),
+					)
+				}
 				for j, actualColResult := range actualRowResult {
 					expectedColResult := expectedRowResult[j]
 					switch expectedColResult {
@@ -403,8 +417,10 @@ func (te tenantExpected) validate(
 				}
 			}
 		} else {
-			require.Errorf(t, err, message)
-			require.Containsf(t, err.Error(), expectedErrorMessage, message)
+			if !testutils.IsError(err, expectedErrorMessage) {
+				// nolint:errwrap
+				return errors.Errorf("%s expected error %q, got %+v", message, expectedErrorMessage, err)
+			}
 		}
 		return nil
 	})

